### ISSUE TITLE
Consider adding a cache snapshot + recover feature

### ISSUE THREAD

**Issue by zoharby** [hoytech/vmtouch#63]
For server applications with a large HDD backed DB, with data >> RAM, it is often the case that the server takes a very long time to "warm up", as it slowly loads the most frequently queried data into RAM in random-access pattern.
The end result after the server is running for some time, is many partly-cached files (which can't fully fit into RAM).
It would be nice to have 2 options:

1. An option to "snapshot" the state of cached blocks under a certain directory into a file
2. Another option to use such a file (after reboot) to load the same blocks into the cache

Since bulk loading the data into cache is done in a much more sequential pattern, this could really shorten the time it takes the server to "warm up", and allow one to only route requests to the server after its cache is filled with the needed data.

**Comment by zoharby** [comment]
Hmm, looks like that exactly what [vmprobe](https://vmprobe.com/) does, so I guess we can close this...

**Comment by hoytech** [comment]
Yes, you're right vmprobe does that, although I'd like to backport the feature to vmtouch since it is much more widely used. Let's leave this open and hopefully I (or someone else) will get around to it. Thanks!

**Comment by AdrienLeBerre** [comment]
Hi,
First of all, thanks for the tool, and sorry to unearth a thread last commented 7 years ago.
I'd like to add a new use case for that feature : I have a rather old server with a lot of RAM but no SSD, and use vmtouch to "preheat" a bunch of directories on the HDD at boot. However, since those files are crawled all around the HDD with no control on the order or wether they will be useful, it can be quite long. Ideally, being able to dump the sorted table of all cached files with their inode, sorted by inode, and grouped by drive, and run vmtouch on it, should ensure that the HDD is accessed as sequencially as possible, ensuring the best loading time, and limit the amount of files that are to be preloaded (for now I do it rather large scale : preload all /bin, /usr/bin, /usr/lib and so on...)
